src/controllers/donationFormController.ts

```diff
@@ -88,52 +88,88 @@ export const postDonationForm = (req: Request, res: Response) => {
-            User.findById(userid)
-        ]
-    ).then((values) => {
-        try {
-            // Parse the 'data' from the request body to get the new donation form
-            const newDonationForm = JSON.parse(req.body.data);
+    // Set up transaction session
+    const session = await mongoose.startSession();
+    session.startTransaction();
 
-            let currentUser:UserDocument;
-            [newDonationForm.imageLink, currentUser] = values; // values is [imagelink, currentuser]
+    try {
+        // Handle image upload and finding specified user with async/await
+        let url = '';
+        if (req.files !== null && req.files !== undefined && req.files.image !== undefined) {
+            url = await uploadImageAzure(req.files.image as UploadedFile);
+        }
 
-            currentUser.donations.push(newDonationForm);
-            currentUser.save().then((updatedUser: UserDocument) => {
-                // Send a push notification to all the admins notifying them of the new donation
-                User.find({ isAdmin: true }).select('pushTokens').exec((err, users) => {
-                    if (err) {
-                        res.status(500).json({ message: err.message, donationform: {} });
-                        return;
-                    }
-                    let tokens:string[] = [];
-                    // Collect the tokens of all the admins
-                    for (const user of users) {
-                        tokens = tokens.concat(user.pushTokens);
-                    }
+        const currentUser = await User.findById(userid).session(session);
 
-                    sendBatchNotification(`New Donation from ${currentUser.businessName}`,
-                        req.body.description ?? 'Check app for details',
-                        tokens);
-                });
-                res.status(200).json({
-                    message: 'Success',
-                    donationform: updatedUser.donations[updatedUser.donations.length - 1]
-                });
-            }).catch((err: Error) => {
-                res.status(500).json({ message: err.message, donationform: {} });
+        // Processing JSON data payload
+        const newDonationForm = JSON.parse(req.body.data);
+        newDonationForm.imageLink = url;
+
+        // Adding new donation to specified user
+        currentUser.donations.push(newDonationForm);
+        const savedDonation = await currentUser.save()
+            .then((updatedUser: UserDocument) => {
+                const donationId = updatedUser.donations[updatedUser.donations.length - 1]._id;
+                newDonationForm._id = donationId;
+                return updatedUser;
             });
-        } catch (err) {
-            res.status(400).json({ message: err.message, donationform: {} });
+
+        // UploadImage and Query User simultaneously by creating a promise out of both asynchronous tasks
+        // Promise.all(
```

Comment:
Use these statements as guide to parallelize the two sequential independant awaits